File: backend/db/Core/Dto/ExerciseDto.cs
using Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Dto
{
public record ExerciseDto(string Name, string Creator, string Description, string Language, List<Tag> Tags, SnippetDto[] arrayOfSnippets, DateTime DateCreated, DateTime DateUpdated);

}

File: frontend/leocode-frontend/src/app/model/exerciseDto.ts
import { CodeSection} from './code-sections';
import { User } from './user';
export interface ExerciseDto {
    name:string,
    creator:string,
    description:string,
    language:string,
    tags:string[],
    arrayOfSnippets: CodeSection[]
    dateCreated: Date,
    dateUpdated: Date,
    teacher: User | undefined,
}